src/intrinsics/ecma262/Object.js

```diff
@@ -131,6 +130,29 @@ export default function(realm: Realm): NativeFunctionValue {
         invariant(frm, "from required");
         invariant(keys, "keys required");
         copyKeys(keys, frm, to);
+      };
+
+      // 4. For each element nextSource of sources, in ascending index order,
+      for (let nextSource of sources) {
+        try {
+          handleSource(nextSource);
```

Comment:
Any error recovery like this needs to evaluate for effects so that they can be reverted. Otherwise we'll end up with the effects up until the fatal and then replaying it again.

Reply:
Will do that now, thanks.

Reply:
Actually, I just tried this and the effects were reverted fine – as we're already in a recovery (from CallExpression).

Reply:
I'm dubious as well.